src/v2i-hub/TelematicBridgePlugin/src/TelematicUnit.h

```diff
@@ -0,0 +1,218 @@
+        vector<string> _selectedTopics;                                       // Global variable to store selected topics confirmed by users
+        static CONSTEXPR const char *AVAILABLE_TOPICS = ".available_topics";  // NATS subject to pub/sub available topics
+        static CONSTEXPR const char *REGISTER_UNIT_TOPIC = "*.register_unit"; // NATS subject to pub/sub registering unit
+        static CONSTEXPR const char *PUBLISH_TOPICS = ".publish_topics";      // NATS subject to publish data stream
+        static CONSTEXPR const char *CHECK_STATUS = ".check_status";          // NATS subject to pub/sub checking unit status
+        natsConnection *_conn = nullptr;                                      // Global NATS connection object
+        natsSubscription *_subAvailableTopic = nullptr;                       // Global NATS subscription object
+        natsSubscription *_subSelectedTopic = nullptr;                        // Global NATS subscription object
+        natsSubscription *_subCheckStatus = nullptr;                          // Global NATS subscription object
+        int64_t TIME_OUT = 10000;                                             // NATS Connection time out in  milliseconds
+        string _eventName;                                                    // Testing event the unit is assigned to
+        string _eventLocation;                                                // Testing event location
+        string _testingType;                                                  // Testing type
+        static CONSTEXPR const char *LOCATION = "location";                   // location key used to find location value from JSON
+        static CONSTEXPR const char *TESTING_TYPE = "testing_type";           // testing_type key used to find testing_type value from JSON
+        static CONSTEXPR const char *EVENT_NAME = "event_name";               // event_name key used to find event_name value from JSON
+        static CONSTEXPR const char *UNIT_ID = "unit_id";                     // unit_id key used to find unit_id value from JSON
+        static CONSTEXPR const char *UNIT_NAME = "unit_name";                 // unit_name key used to find unit_name value from JSON
+        static CONSTEXPR const char *UNIT_TYPE = "unit_type";                 // unit_type key used to find unit_type value from JSON
+        static CONSTEXPR const char *TOPIC_NAME = "topic_name";               // topic_name key used to find topic_name value from JSON
+        static CONSTEXPR const char *TIMESTAMP = "timestamp";                 // timestamp key used to find timestamp value from JSON
+        static CONSTEXPR const char *PAYLOAD = "payload";                     // payload key used to find payload value from JSON
+        static CONSTEXPR const char *TOPICS = "topics";                       // topics key used to find topics value from JSON
```

Comment:
These are not topics, I would refrain from using this language unless telematics requires it.

Reply:
I can change the variable name "TOPICS". We do need the value topics to find JSON value that match this topics key.